bcipy/helpers/triggers.py

```diff
@@ -145,61 +145,66 @@ def _write_triggers_from_inquiry_calibration(array: list,
     return trigger_file
 
 
-def _write_triggers_from_inquiry_copy_phrase(array,
-                                             trigger_file,
-                                             copy_text,
-                                             typed_text,
-                                             offset=None):
-    """
-    Write triggers from copy phrase.
-
-    Helper Function to write trigger data to provided trigger_file. It assigns
-        target letter based on matching the next needed letter in typed text
-        then assigns target/nontarget label to following letters.
+def _write_triggers_from_inquiry_copy_phrase(
+        triggers: list,
+        trigger_file: TextIO,
+        copy_phrase: str,
+        typed_text: str,
+        offset: bool = False) -> TextIO:
+    """Write triggers from copy phrase.
+
+    Helper Function to write trigger data to provided trigger_file in a copy phrase task.
+        It assigns target letter based on matching the next needed letter in typed text
+        then assigns target/nontarget label to following letters. It's assumed if offset
+        is true, only one record will be written (the offset_correction).
 
     It writes in the following order:
         (I) presented letter, (II) targetness, (III) timestamp
     """
-
+    # write the offset and return the file
     if offset:
         # extract the letter and timing from the array
-        (letter, time) = array
-        targetness = 'offset_correction'
-        trigger_file.write('%s %s %s' % (letter, targetness, time) + "\n")
-
+        (symbol, time) = triggers
+        trigger_file.write(f'{symbol} offset_correction {time}\n')
+        return trigger_file
+
+    spelling_length = len(typed_text)
+    last_typed = typed_text[-1] if typed_text else None
+    target_symbol = copy_phrase[spelling_length - 1]
```

Comment:
The copy_phrase task already computes the target symbol. I wonder if that should be provided as a parameter rather than computing it here. Or maybe we should factor out that code into a helper function. 

Also, with the copy phrase task method (`next_target`) we looked at the all of the spelled text, rather than just the last letter. That way if the phrase is 'HELLO' and they have spelled 'HEEL' the target would be backspace, rather than 'O'.

Reply:
This is an interesting point... I think we should talk more with the clinical team on how they determine correctness or targetness in this case. If we were going to report on copy phrase performance, for instance, would HEELO be 80% or  40% correct because there was no correction? Would then HEE<LLO be ~85%?

Reply:
I think this would be an interesting discussion for the language modeling team as well. It seems like the answer could depend on whether we're using letter vs word-based probabilities.